src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -472,6 +487,12 @@ def generate_config(
         }
 
         if self.mgr.secure_monitoring_stack:
+            r2: Dict[str, Any] = {'files': {}}
+            for url, credentials in clusters_credentials.items():
+                r2['files'][f'prometheus_{url}_cert.crt'] = credentials['certificate']
+                credentials['cert_file_name'] = f'prometheus_{url}_cert.crt'
+                context['clusters_credentials'] = clusters_credentials
+
             cfg_key = 'mgr/prometheus/root/cert'
             cmd = {'prefix': 'config-key get', 'key': cfg_key}
             ret, mgr_prometheus_rootca, err = self.mgr.mon_command(cmd)
```

Comment:
given you've added a constant for where to get the cert from, can we modify this to use `get_store`?

Reply:
I didn't get this completely

Reply:
never mind, this is slightly different than the constant you added. We probably should add one for it, but that's irrelevant to this PR